src/gleam/list.gleam

```diff
@@ -1565,6 +1565,50 @@ pub fn each(list: List(a), f: fn(a) -> b) -> Nil {
   }
 }
 
+/// A variant of each that might fail.
+///
+/// Takes a function that returns a result and tries to calls it for each element in a list.
+/// If the returned value is `Ok(new_value)`, try_each will discard the return value and call
+/// the function on the next element of the list.
+/// If the returned value is `Error(error)`, try_each will stop and return Nil.
+///
+/// ## Examples
+///
+/// ```gleam
+/// list.try_each(
+///   over: [1, 2, 3],
+///   with: fn(x) {
+///     io.print(int.to_string(x))
+///     Ok(Nil)
+///   },
+/// )
+/// // -> 123
```

Comment:
the convention so far is prefixing each syntax line with a `> ` and then just adding the return value as the last line:

Example:

```
/// > reverse([1, 2])
/// [2, 1]
```

also `// -> 123` seems wrong `Nil` should it be.